### ISSUE TITLE
[BUG] mob music on wrong mobs due to radius

### ISSUE THREAD

**Issue by TheRealZonko** [viniciusgf2/Ambience#41]
So in 1.12.2, I made it so mob music happens in a 40 block radius with a specific mob rather than 12 blocks, but whenever I get hit by a mob, other mobs that could play their music theme within that radius begin to play it.  I have quite a few underground bosses that naturally spawn and it gets really weird when their music plays when I get hit by a zombie.

Hope you are able to fix this man!

**Comment by viniciusgf2** [comment]
yes, this will happen if you put a great radius, because there isn't an easy way to know who attacked the player so it scans every mob in that radius check if is hostile and get the first mob of the list that usually are the ones that is attacking you, because of that the default radius is short, to solve this I would ask to the server who attacked the player and etc and this will cause performance issues
